Assignment/test.c

```diff
@@ -0,0 +1,110 @@
+#include <stdlib.h>
+#include <limits.h> 
+#include "conversion.h"
+
+#define INVALID_INPUT "Invalid Input"
+// ANSI color code for green and red text
+#define GREEN_TEXT "\033[0;32m"
+#define RED_TEXT "\033[0;31m"
+#define RESET_TEXT "\033[0m"
+
+int IsValid (const char* n) {
+   if (n == NULL || *n == '\0') return 0;
+   char* endptr;
+   long num = strtol (n, &endptr, 10);
+   if (*endptr != '\0' || num < INT_MIN || num > INT_MAX) return 0;
+   return 1;
+}
+
+void PrintResult (const char* result) {
+   int padding = 100 - (int)strlen (result) - 1;
+   for (int i = 0; i < padding; i++) printf (" ");
+   printf ("%s\n", result);
+}
+
+void DecToBinaryTest () {
+   char* input[] = { "0", "5", "255", "-255", "2147483647", "-2147483647", "*", "abc56" };
+   char* expected_output[] = { "00000000", "00000101", "0000000011111111", "1111111100000001",
+                               "01111111111111111111111111111111", "10000000000000000000000000000001",
+                               "Invalid Input", "Invalid Input" };
+   int testCount = sizeof (input) / sizeof (input[0]);
+   printf ("---Decimal to Binary---\n\n");
+   printf ("| %-12s | %-35s | %-40s | %-19s |\n", "Input", "Expected Output", "Actual Output", "Test Case Result");
+   printf ("|--------------|-------------------------------------|------------------------------------------|---------------------|\n");
+   for (int i = 0; i < testCount; i++) {
+      char* actualBinary = IsValid (input[i]) ? DecToBinary (atoi (input[i])) : INVALID_INPUT;
+      const char* result = (strcmp (actualBinary, expected_output[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
+      printf ("| %-12s | %-35s | %-40s | %-30s |\n", input[i], expected_output[i], actualBinary, result);
+   }
+   printf ("\n\n");
+}
+
+void DecToHexaTest () {
+   char* input2[] = { "0", "5", "255", "-255", "2147483647", "-2147483647", "*", "vini" };
+   char* expected_output[] = { "00", "05", "00FF", "FF01","7FFFFFFF", "80000001","Invalid Input", 
+                               "Invalid Input" };
+   int testCount = sizeof (input2) / sizeof (input2[0]);
+   printf ("---Decimal to Hexadecimal---\n\n");
+   printf ("| %-12s | %-20s | %-20s | %-20s |\n", "Input", "Expected Output", "Actual Output", "Test Case Result");
+   printf ("|--------------|----------------------|----------------------|----------------------|\n");
+   for (int i = 0; i < testCount; i++) {
+      char* actualHexa = IsValid (input2[i]) ? DecToHexa (atoi (input2[i])) : INVALID_INPUT;
+      const char* result = (strcmp (actualHexa, expected_output[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
+      printf ("| %-12s | %-20s | %-20s | %-31s |\n", input2[i], expected_output[i], actualHexa, result);
+   }
+}
+
+void UserInput () {
+   long long number;
+   char nextChar;
+   while (1) {
+      printf ("\nEnter a decimal number: ");
+      if (scanf_s ("%lld%c", &number, &nextChar, 1) != 2 || nextChar != '\n') {

```

Comment:
It is not recommended to use scanf to get user inputs since you know the input may not be formatted. 
Your program breaks if I enter "gggh" or anything greater than 1 char in length.
![image](https://github.com/user-attachments/assets/d3b23e3d-8a1e-4a1a-b698-51cfa75d2397)

Reply:
You are checking if your input is valid for test cases but that is not reflected in your actual program